src/Engine/Rendering/Lighting/DirectionalLight.cpp

```diff
@@ -16,27 +16,10 @@ namespace Engine
 		namespace Lighting
 		{
 			float ortho_size = 3;
-			DirectionalLight::DirectionalLight() :
-				intensity(1.0f),
-				// TODO: figure out the best value for this
-				depthbuffer(DepthBuffer(200, 200)),
-				projection_mat(Matrix4().SetOrthographicProjection(-ortho_size, ortho_size, ortho_size, -ortho_size, -ortho_size, ortho_size))
-			{
-				// by calling the setter explicitly we also update the view matrix
-				SetDirection(Vector3());
-			}
-
-			DirectionalLight::DirectionalLight(const Vector3& direction) :
-				intensity(1.0f),
-				// TODO: figure out the best value for this
-				depthbuffer(DepthBuffer(200, 200)),
-				projection_mat(Matrix4().SetOrthographicProjection(-ortho_size, ortho_size, ortho_size, -ortho_size, -ortho_size, ortho_size))
-			{
-				SetDirection(direction);
-			}
 
-			DirectionalLight::DirectionalLight(const Vector3& direction, float intensity) :
+			DirectionalLight::DirectionalLight(const Vector3& direction, float intensity, RGBColor color) :
 				intensity(intensity),
+				color(RGBColor()),
```

Comment:
Why isn't intensity just the length of the color?

Reply:
Cause I didn't think about that, nice idea 👀 :D

Reply:
And regarding the first comment, right now I really dislike how light transformations space are done in the shader along with how the light depth buffer is passed for shadow maps

I think it's a problem that needs to be fixed eventually, but I still like having a shader independent lighting system, where you can add and remove lights, set ambient lighting etc...IMO it's up to the the shader how it wants to allow the lights to influence it's results (if at all), but doesn't have to manage the lights or hold them